src/cargo.rs

```diff
@@ -432,6 +434,88 @@ pub fn rustdoc_flags(config: &Config) -> String {
     value
 }
 
+fn look_for_rustflags_in_table(value: &Value) -> String {
+    let table = value.as_table().unwrap();
+
+    if let Some(rustflags) = table.get("rustflags") {
+        let vec_of_flags: Vec<String> = rustflags
+            .as_array()
+            .unwrap()
+            .into_iter()
+            .filter_map(|x| x.as_str())
+            .map(|x| x.to_string())
+            .collect();
+
+        vec_of_flags.join(" ")
+    } else {
+        String::new()
+    }
+}
+
+fn look_for_rustflags_in_file(path: &Path) -> Option<String> {
+    if let Ok(contents) = read_to_string(path) {
+        let value = contents.parse::<Value>().ok()?;
+
+        let rustflags_in_file: Vec<String> = value
+            .as_table()?
+            .into_iter()
+            .map(|(s, v)| {
+                if s.as_str() == "build" {
+                    look_for_rustflags_in_table(v)
+                } else {
+                    String::new()
+                }
+            })
+            .collect();
+
+        Some(rustflags_in_file.join(" "))
+    } else {
+        None
+    }
+}
+
+fn look_for_rustflags_in(path: &Path) -> Option<String> {
+    let mut config_path = PathBuf::from(&path);
+    config_path.push("config");
```

Comment:
Just realised you can simplify this a bit by doing `let mut config_path = path.join("config");`